src/tiledb/cloud/vcf/ingestion.py

```diff
@@ -611,29 +635,37 @@ def filter_samples_udf(
                 dataset_uri,
                 cfg=tiledbvcf.ReadConfig(tiledb_config=config),
             )
-            existing_samples = set(ds.samples())
+            dataset_samples = set(ds.samples())
 
-            # Read all samples in the manifest with status == "ok"
+            # Read all samples in the manifest and samples ready for ingestion
             group = tiledb.Group(dataset_uri)
             manifest_uri = group[MANIFEST_ARRAY].uri
 
+            cond = f"status == '{Status.READY}' or status == '{Status.MISSING_INDEX}'"
             with tiledb.open(manifest_uri) as A:
-                manifest_df = A.query(
-                    cond="status == 'ok' or status == 'missing index'"
-                ).df[:]
+                manifest_df = A.df[:]
+                ingest_df = A.query(cond=cond).df[:]
+            manifest_samples = set(manifest_df.sample_name)
 
             # Sort manifest by sample_name
-            manifest_df = manifest_df.sort_values(by=["sample_name"])
+            ingest_df = ingest_df.sort_values(by=["sample_name"])
+            ingest_samples = set(ingest_df.sample_name)
 
-            # Find samples that have not already been ingested
-            manifest_samples = set(manifest_df.sample_name)
-            new_samples = manifest_samples.difference(existing_samples)
-            manifest_df = manifest_df[manifest_df.sample_name.isin(new_samples)]
-            result = manifest_df.vcf_uri.to_list()
+            # Find samples that have already been ingested or failed to ingest
+            existing_samples = dataset_samples.difference(ingest_samples)
+            failed_samples = dataset_samples.intersection(ingest_samples)
+
+            # Finalize samples to ingest
+            if not resume:
+                ingest_df = ingest_df[~ingest_df.sample_name.isin(failed_samples)]
+            result = ingest_df.vcf_uri.to_list()
 
+            logger.info("%d samples in the dataset.", len(dataset_samples))
```

Comment:
@alancleary what would you think about making the log messages more structured. I don't mean using [structlog](https://www.structlog.org/en/stable/) necessarily, but changing the messages from being prose to being `event` + `data`. For example: `logger.info("Ingestion iteration completed: successful_samples=%r, failed_samples=%r", ...)`. Everything on a single line.

Messages like these are easy to scan for and can be more easily parsed and turned into data.

I know that's not the prevailing style in our Python code, but I'm trying to nudge new work in the structured direction.

Reply:
Thanks for the suggestion @sgillies. That makes sense to me. I'll make the change.

Reply:
Here are a couple of examples:

* https://github.com/TileDB-Inc/Internal-Cloud-Py/blob/7a3f33097ee05ab14508ea0c4eb06d9fb5120d8f/src/tiledb/cloud/taskgraphs/server_executor/impl.py#L213
* https://github.com/TileDB-Inc/Internal-Cloud-Py/blob/7a3f33097ee05ab14508ea0c4eb06d9fb5120d8f/src/tiledb/cloud/soma/ingest.py#L112

Reply:
Thanks @sgillies. Looking at the logging code now, reworking all the messages in ingestion would require touching a bit of code unrelated to this PR so I think I'll do it as its own PR or as part of a manifest rework. Are there guidelines anywhere for this that I can reference in an issue?

Reply:
@alancleary I'm only asking about the new logging you've added. We can revisit the existing code later, I agree.

We don't have any guidelines about logging, but I'll add something to a contributing doc.

Reply:
@alancleary Yes, like that. In looking around for good examples on the web, I came across this one from Stripe: https://brandur.org/canonical-log-lines#what-are-they. "Literally one big log line".

I think you could leverage f-strings to simplify the above.

```python
logger.info(f"Filtering sample URIs: {len(dataset_samples)=}, {len(ingested_samples)=}, {len(partial_samples)=}, {len(manifest_samples)=}, {len(ready_samples)=}, {len(queued_samples)=}")
```

With the `=` syntax, you get, for example, `"Filtering sample URIs: len(dataset_samples)=100000, ..."`.

Reply:
> "Literally one big log line"

I like it! Also, I haven't encountered `=` syntax with f-strings before. Thanks for sharing!